spec/index.html

```diff
@@ -1493,3 +1503,221 @@ <h1>SharedArray ( ..._values_ )</h1>
+    </ul>
+
+    <emu-note>
+      <p>Per-Realm prototypes, which is currently an open design question and not included in this draft, will give Mutexes a per-Realm prototype with built-in methods instead of static methods, and will also add support for the %Symbol.dispose%.</p>
+    </emu-note>
+
+    <emu-clause id="sec-atomics.mutex.lock">
+      <h1>Atomics.Mutex.lock ( _mutex_ [ , _unlockToken_ [ , _timeout_ ] ] )</h1>
+      <p>This function puts the surrounding agent in a wait queue and suspends it until the mutex is unlocked or until the wait times out, returning a String differentiating those cases.</p>
+      <p>It performs the following steps when called:</p>
+      <emu-alg>
+        1. If _mutex_ is not an Object, throw a *TypeError* exception.
+        1. If _mutex_ does not have a [[MutexWaiterList]] internal slot, throw a *TypeError* exception.
+        1. If _unlockToken_ not *undefined*, then
+          1. If _unlockToken_ is not an Object, throw a *TypeError* exception.
+          1. If _unlockToken_ does not have a [[LockedMutex]] internal slot, throw a *TypeError* exception.
+          1. If _unlockToken_.[[LockedMutex]] is not ~empty~, throw a *TypeError* exception.
+        1. If _timeout_ is *undefined*, set _timeout_ to *+∞*<sub>𝔽</sub>.
+        1. If _timeout_ is not a Number, throw a *TypeError* exception.
+        1. If _timeout_ is either *NaN* or *+∞*<sub>𝔽</sub>, let _t_ be +∞; else if _timeout_ is *-∞*<sub>𝔽</sub>, let _t_ be 0; else let _t_ be max(ℝ(_timeout_), 0).
+        1. If AgentCanSuspend() is *false*, throw a *TypeError* exception.
+        1. Let _result_ be LockMutex(_mutex_, _t_).
+        1. If _result_ is ~acquired~, then
+          1. Return UnlockTokenCreateIfNeeded(_unlockToken_, _mutex_).
+        1. Else,
+          1. Assert: _result_ is ~timed-out~.
+          1. Return *"timed-out"*.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-atomics.mutex.trylock">
+      <h1>Atomics.Mutex.tryLock ( _mutex_ [ , _unlockToken_ ] )</h1>
+      <p>It performs the following steps when called:</p>
+      <emu-alg>
+        1. If _mutex_ is not an Object, throw a *TypeError* exception.
+        1. If _mutex_ does not have a [[MutexWaiterList]] internal slot, throw a *TypeError* exception.
+        1. If _unlockToken_ not *undefined*, then
+          1. If _unlockToken_ is not an Object, throw a *TypeError* exception.
+          1. If _unlockToken_ does not have a [[LockedMutex]] internal slot, throw a *TypeError* exception.
+          1. If _unlockToken_.[[LockedMutex]] is not ~empty~, throw a *TypeError* exception.
+        1. Let _WL_ be _mutex_.[[MutexWaiterList]].
+        1. Perform EnterCriticalSection(_WL_).
+        1. If _mutex_.[[IsLocked]] is *false*, then
+          1. Set _mutex_.[[IsLocked]] to *true*.
+          1. Let _result_ be UnlockTokenCreateIfNeeded(_unlockToken_, _mutex_).
+        1. Else,
+          1. Let _result_ be *undefined*.
```

Comment:
```suggestion
          1. Let _result_ be *"already-locked"*.
```
both to match the `lock` return type and to avoid the problem with `using` mentioned in the previous comment.